numpy/core/tests/test_multiarray.py

```diff
@@ -89,6 +89,24 @@ def test_writeable(self):
         self.a[0] = 5
         self.a[0] = 0
 
+        # gh-9440 - make sure fromstring, from buffer set writeable False
+        data = b'\x00' * 100
+        vals = np.frombuffer(data, 'B')
+        assert_raises(ValueError, vals.setflags, write=True)
+        types = np.dtype( [('vals', 'u1'), ('res3', 'S4')] )
+        values = np.core.records.fromstring(data, types)
```

Comment:
Looks like we ought to rename this to `frombuffer` at some point, especially once we only support python 3 and `fromstring(str)` fails but `fromstr(bytes)` succeeds.